package in.co.rays.proj4.controller;

import in.co.rays.proj4.bean.MarksheetBean;
import in.co.rays.proj4.model.MarksheetModel;
import in.co.rays.proj4.util.DataUtility;
import jakarta.servlet.annotation.WebServlet;
import jakarta.servlet.http.HttpServletRequest;

@WebServlet("/ctl/MarksheetListCtl")
public class MarksheetListCtl
		extends BaseListCtl<MarksheetBean, MarksheetModel> {

	@Override
	protected MarksheetBean populateBean(
			HttpServletRequest request) {

		MarksheetBean bean =
				new MarksheetBean();

		bean.setRollNo(
				DataUtility.getString(
						request.getParameter("rollNo")));

		bean.setStudentId(
				DataUtility.getLong(
						request.getParameter("studentId")));

		bean.setName(
				DataUtility.getString(
						request.getParameter("name")));

		return bean;
	}


	@Override
	protected String getView() {

		return ORSView.MARKSHEET_LIST_VIEW;
	}


	@Override
	protected MarksheetModel getModel() {

		return new MarksheetModel();
	}
}